Mathlib/Data/Nat/Choose/Sum.lean

```diff
@@ -33,114 +29,116 @@ variable [Semiring R] {x y : R}
-      ∑ i ∈ range (m + 1), choose (2 * m + 1) i :=
+theorem sum_range_choose_halfway (m : ℕ) : (∑ i ∈ range (m + 1), (2 * m + 1).choose i) = 4 ^ m :=
+  have : (∑ i ∈ range (m + 1), (2 * m + 1).choose (2 * m + 1 - i)) =
+      ∑ i ∈ range (m + 1), (2 * m + 1).choose i :=
     sum_congr rfl fun i hi ↦ choose_symm <| by linarith [mem_range.1 hi]
   mul_right_injective₀ two_ne_zero <|
     calc
-      (2 * ∑ i ∈ range (m + 1), choose (2 * m + 1) i) =
-          (∑ i ∈ range (m + 1), choose (2 * m + 1) i) +
-            ∑ i ∈ range (m + 1), choose (2 * m + 1) (2 * m + 1 - i) := by rw [two_mul, this]
-      _ = (∑ i ∈ range (m + 1), choose (2 * m + 1) i) +
-            ∑ i ∈ Ico (m + 1) (2 * m + 2), choose (2 * m + 1) i := by
-        { rw [range_eq_Ico, sum_Ico_reflect]
-          · congr
-            have A : m + 1 ≤ 2 * m + 1 := by omega
-            rw [add_comm, add_tsub_assoc_of_le A, ← add_comm]
-            congr
-            rw [tsub_eq_iff_eq_add_of_le A]
-            ring
-          · omega }
-      _ = ∑ i ∈ range (2 * m + 2), choose (2 * m + 1) i := sum_range_add_sum_Ico _ (by omega)
+      (2 * ∑ i ∈ range (m + 1), (2 * m + 1).choose i) =
+          (∑ i ∈ range (m + 1), (2 * m + 1).choose i) +
+            ∑ i ∈ range (m + 1), (2 * m + 1).choose (2 * m + 1 - i) := by rw [two_mul, this]
+      _ = (∑ i ∈ range (m + 1), (2 * m + 1).choose i) +
+            ∑ i ∈ Ico (m + 1) (2 * m + 2), (2 * m + 1).choose i := by
+        rw [range_eq_Ico, sum_Ico_reflect _ _ (by omega)]
+        congr
+        have A : m + 1 ≤ 2 * m + 1 := by omega
+        rw [add_comm, add_tsub_assoc_of_le A, ← add_comm]
+        congr
+        rw [tsub_eq_iff_eq_add_of_le A]
+        ring
+      _ = ∑ i ∈ range (2 * m + 2), (2 * m + 1).choose i := sum_range_add_sum_Ico _ (by omega)
       _ = 2 ^ (2 * m + 1) := sum_range_choose (2 * m + 1)
-      _ = 2 * 4 ^ m := by rw [Nat.pow_succ, pow_mul, mul_comm]; rfl
+      _ = 2 * 4 ^ m := by rw [pow_succ, pow_mul, mul_comm]; rfl
 
-theorem choose_middle_le_pow (n : ℕ) : choose (2 * n + 1) n ≤ 4 ^ n := by
-  have t : choose (2 * n + 1) n ≤ ∑ i ∈ range (n + 1), choose (2 * n + 1) i :=
+theorem choose_middle_le_pow (n : ℕ) : (2 * n + 1).choose n ≤ 4 ^ n := by
+  have t : (2 * n + 1).choose n ≤ ∑ i ∈ range (n + 1), (2 * n + 1).choose i :=
     single_le_sum (fun x _ ↦ by omega) (self_mem_range_succ n)
   simpa [sum_range_choose_halfway n] using t
 
 theorem four_pow_le_two_mul_add_one_mul_central_binom (n : ℕ) :
-    4 ^ n ≤ (2 * n + 1) * choose (2 * n) n :=
+    4 ^ n ≤ (2 * n + 1) * (2 * n).choose n :=
   calc
-    4 ^ n = (1 + 1) ^ (2 * n) := by norm_num [pow_mul]
-    _ = ∑ m ∈ range (2 * n + 1), choose (2 * n) m := by set_option simprocs false in simp [add_pow]
-    _ ≤ ∑ m ∈ range (2 * n + 1), choose (2 * n) (2 * n / 2) := by gcongr; apply choose_le_middle
-    _ = (2 * n + 1) * choose (2 * n) n := by simp
+    _ = (1 + 1) ^ (2 * n) := by norm_num [pow_mul]
```

Comment:
In calc blocks, it's often more readable to keep the first and last entry instead of putting underscores. So
```suggestion
    4 ^ n = (1 + 1) ^ (2 * n) := by norm_num [pow_mul]
```